resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -284,6 +284,24 @@ private[spark] object Config extends Logging {
       .toSequence
       .createWithDefault(Nil)
 
+  val KUBERNETES_EXECUTOR_POD_DECOMMISSION_LABEL =
+    ConfigBuilder("spark.kubernetes.executor.pod.decommmissionLabel")
+      .doc("Label to apply to a pod which is being decommissioned." +
+        " Designed for use with pod disruption budgets and similar mechanism " +
+        " such as pod-deletion-cost.")
+      .version("3.2.0")
+      .stringConf
+      .createOptional
+
+  val KUBERNETES_EXECUTOR_POD_DECOMMISSION_LABEL_VALUE =
+    ConfigBuilder("spark.kubernetes.executor.pod.decommmissionLabelValue")
+      .doc("Label value to apply to a pod which is being decommissioned." +
+        " Designed for use with pod disruption budgets and similar mechanism " +
+        " such as pod-deletion-cost.")
+      .version("3.2.0")
+      .stringConf
+      .createWithDefault("")
```

Comment:
Shall we use `createOptional`?

Reply:
So my thought was that for a labels value we'd use an empty string if the user doesn't specify a value, but I'm down for something else.

Reply:
Yes, we can still attach `LABEL` without values with `createOptional` conf.

Reply:
So I don't have strong feelings about this one, but the label does need a value (albiet the empty string is a valid value), so I've switched to use optional and then we do a getOrElse on it but I can switch it back.